### ISSUE TITLE
Batch reconcile problem

### ISSUE THREAD

**Issue by tomwaldnz** [hlandau/acme#227]
Yesterday my server emailed me that the acmetool batch reconcile had failed. The certificate is valid for at least a month so it's not urgent, but that seems like something I should look at. Server is Amazon Linux.

Here's the email contents

```
20161127061703 [ERROR] acme.storageops: could not obtain authorization for example.com: failed all combinations
20161127061703 [ERROR] acme.storageops: Target(example.com,www.example.com;https://acme-v01.api.letsencrypt.org/directory;0): failed to request certificate: failed all combinations
```

I ran the command "/usr/local/bin/acmetool  --xlog.severity=debug" and got a lot of output, none of which I understood. I could post the output somewhere if I could get the output to pipe to a file, which I couldn't with either | or >

 I ran "acmetool quickstart" but it looked like it wanted to set things up from scratch, which would be take a fair bit of time given the four domain I have, and it took a while the first time.

Any suggestions how to resolve would be appreciated.

**Comment by Mrten** [comment]
Run `acmetool --xlog.severity=debug > /tmp/dump 2>&1` to get output in a `/tmp/dump`. Then grep away `fdb` and `storageops` like so: `fgrep -v fdb: /tmp/dump | fgrep -v storageops: | less` and you'll probably be left with the actual problems.

Sometimes some interpreting is necessary; for example, when the user isn't right on the webroot dir and acmetool can't write the challenges you won't get a specific error, you'll have to search around a bit.

**Comment by tomwaldnz** [comment]
Thanks to @Mrten I've worked out the problem: acme is writing the challenge file to one directory, and Let's Encrypt is requesting from another directory. The problem was my Nginx configuration - I was using root instead of alias. Root tells Nginx to append the directories in the request, alias tells it to just look for the filename requested ignoring paths.

This works

```
 location ^~ /.well-known/acme-challenge/ {
    alias /var/www/acme-challenge/;
  }
```

This doesn't

```
 location  /.well-known/acme-challenge/ {
    root /var/www/acme-challenge/;
  }
```

For anyone who has this problem in future here's how I followed this through to solve the problem. The acme logs showed the challenge files were being written, so the question was why couldn't Let's Encrypt find them?

I tracked this down looking at the web server logs. I could see that Nginx was looking in the directory "/var/www/acme-challenge/.well-known/acme-challenge", but the logs showed acme was writing to "/var/www/acme-challenge/". It was clear that either acme needed to write to a different directory (couldn't do that, no idea where the config file is) or Nginx needed to look in the right place.

Here's the line that was my major clue (tidied up a bit from the logs)
`open() "/var/www/acme-challenge/.well-known/acme-challenge/p64-Tlqu327rd2E0ALgwx5Fc4WuMTCuybTYMbPWUBWA" failed (2: No such file or directory), request: "GET /.well-known/acme-challenge/p64-Tlqu327rd2E0ALgwx5Fc4WuMTCuybTYMbPWUBWA HTTP/1.1",`

I will have used something like [this post](https://community.letsencrypt.org/t/how-to-nginx-configuration-to-enable-acme-challenge-support-on-all-http-virtual-hosts/5622) to help set up Nginx, but it doesn't appear to work on my configuration. Changing from "root" to "alias" told Nginx not to follow the paths in the request - information about that is [here](http://stackoverflow.com/questions/10631933/nginx-static-file-serving-confusion-with-root-alias).
